### PR TITLE
add cdnendpoint facts module

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

add azure_rm_cdnendpoint_facts.py

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Fred-sun** [comment]
@zikalino @yuwzho review_need

File: lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Hai Cao, <t-haicao@microsoft.com>
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py]: still 2018 and e-mail

File: lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+            https_port=cdnendpoint.origins[0].https_port
+        )
+        new_result['tags'] = cdnendpoint.tags
+        return new_result
+
+
+def main():
+    """Main module execution code path"""
+
+    AzureRMCdnendpointFacts()
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py]: Maybe should be AzureRMCdnEndpointFacts, but that's minor

**Review by zikalino — APPROVED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py (hunk 2 above)
Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py]: will fix it

File: lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py (hunk 1 above)
Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py]: will fix year. email works here, but not -author field

File: test/integration/targets/azure_rm_cdnprofile/tasks/main.yml [hunk 3]
```diff
@@ -143,6 +143,22 @@
...
+      name: "{{ endpointname }}"
+      profile_name: "{{ cdnprofilename }}"
+  register: facts
+
+- name: Assert facts output
+  assert:
+    that:
+      - facts['cdnendpoints'] | length == 1
+      - facts['cdnendpoints'][0]['id']
+      - facts['cdnendpoints'][0]['name']
```
Comment (zikalino) [inline test/integration/targets/azure_rm_cdnprofile/tasks/main.yml:158]: maybe just check all returned fields?
and also try to gather facts without specifying "name" parameter

File: lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+            returned: always
+            type: str
+            sample: myEndpoint
+        profile_name:
+            description:
+                - Name of the Azure CDN profile that this endpoint is attached to.
+            returned: always
+            type: str
+            sample: myProfile
+        state:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint_facts.py]: maybe we can drop state. it was my idea in the beginning, but i think it's not really necessary now as always the same.

**Pull request merged**